pkg/apis/hiveinternal/v1alpha1/clustersync_types.go

```diff
@@ -0,0 +1,121 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterSync is the status of all of the SelectorSyncSets and SyncSets that apply to a ClusterDeployment.
+// +k8s:openapi-gen=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=clustersyncs,scope=Namespaced
+type ClusterSync struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ClusterSyncSpec   `json:"spec,omitempty"`
+	Status ClusterSyncStatus `json:"status,omitempty"`
+}
+
+// ClusterSyncSpec defines the desired state of ClusterSync
+type ClusterSyncSpec struct{}
+
+// ClusterSyncStatus defines the observed state of ClusterSync
+type ClusterSyncStatus struct {
+	// SyncSets is the sync status of all of the SyncSets for the cluster.
+	// +optional
+	SyncSets []SyncStatus `json:"syncSets,omitempty"`
+
+	// SelectorSyncSets is the sync status of all of the SelectorSyncSets for the cluster.
+	// +optional
+	SelectorSyncSets []SyncStatus `json:"selectorSyncSets,omitempty"`
+
+	// Conditions is a list of conditions associated with syncing to the cluster.
+	// +optional
+	Conditions []ClusterSyncCondition `json:"conditions,omitempty"`
+}
+
+// SyncStatus is the status of applying a specific SyncSet or SelectorSyncSet to the cluster.
+type SyncStatus struct {
+	// Name is the name of the SyncSet or SelectorSyncSet.
+	Name string `json:"name"`
+
+	// ObservedGeneration is the generation of the SyncSet or SelectorSyncSet that was last observed.
+	ObservedGeneration int64 `json:"observedGeneration"`
+
+	// ResourcesToDelete is the list of resources in the cluster that should be deleted when the SyncSet or SelectorSyncSet
+	// is deleted or is no longer matched to the cluster.
+	// +optional
+	ResourcesToDelete []hivev1.SyncResourceReference `json:"resourcesToDelete,omitempty"`
+
+	// Result is the result of the last attempt to apply the SyncSet or SelectorSyncSet to the cluster.
+	Result SyncSetResult `json:"result"`
+
+	// FailureMessage is a message describing why the SyncSet or SelectorSyncSet could not be applied. This is only
+	// set when Result is Failure.
+	// +optional
+	FailureMessage string `json:"failureMessage,omitempty"`
+
+	// LastTransitionTime is the time when this status last changed.
```

Comment:
This is the last time that there was an change. This is consistent with the `LastTransitionTime` when used in conditions. I am reticent to rename it to `LastUpdateTime` because I could see that being misinterpreted as that last time that a apply was done. Let me think more on what a better name for this field may be.